### PR TITLE
Add support for registrant APIs

### PR DESCRIPTION
**Pull request by wilson-dnsimple**
This PR introduces 5 new endpoints that are part of the domain contact management API that was introduced in https://github.com/dnsimple/dnsimple-developer/pull/501.

The following endpoints have been added:
- **listRegistrantChanges**: GET `/{account}/registrar/registrant_changes`
- **createRegistrantChange**: POST `/{account}/registrar/registrant_changes`
- **checkRegistrantChange**: POST `/{account}/registrar/registrant_changes/check`
- **getRegistrantChange**: GET `/{account}/registrar/registrant_changes/{registrantchange}`
- **deleteRegistrantChange**: DELETE `/{account}/registrar/registrant_changes/{registrantchange}`

Belongs to https://github.com/dnsimple/dnsimple-business/issues/1729

### CODE REVIEW

File: dnsimple/registrar.go [hunk 1]
```diff
@@ -364,3 +364,46 @@ func (s *RegistrarService) RenewDomain(ctx context.Context, accountID string, do
...
+	changeResponse := &RegistrantChangeResponse{}
+
+	resp, err := s.client.post(ctx, path, input, changeResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	changeResponse.HTTPResponse = resp
+	return changeResponse, nil
+}
```
Comment (DXTimer) [inline dnsimple/registrar.go:409]: We roughly group API code based on the tag. So the new contact change endpoints should go into a separate file perhaps `regisrant_changes.go`.

**Review by DXTimer — CHANGES_REQUESTED** [review]
Only the create registrant change endpoint has been added please add all 5 endpoints for the registrant change API: list, check, create, show, delete.

Tests are missing.

File: dnsimple/registrar_registrant.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+type CreateRegistrantChangeInput struct {
+	DomainId           int               `json:"domain_id"`
+	ContactId          int               `json:"contact_id"`
+	ExtendedAttributes map[string]string `json:"extended_attributes"`
+}
+
+type RegistrantChange struct {
+	Id        int `json:"id"`
+	Type      int `json:"type"`
```
Comment (DXTimer) [inline dnsimple/registrar_registrant.go:16]: This is not an attribute that exists on the RegistrantChange resource.

File: dnsimple/dnsimple.go [hunk 3]
```diff
@@ -262,7 +262,15 @@ func (c *Client) request(ctx context.Context, req *http.Request, obj interface{}
 		if w, ok := obj.(io.Writer); ok {
 			_, err = io.Copy(w, resp.Body)
 		} else {
-			err = json.NewDecoder(resp.Body).Decode(obj)
+			var raw []byte
+			raw, err = io.ReadAll(resp.Body)
+			if err == nil {
+				if len(raw) == 0 {
+					// TODO Ignore empty body as temporary workaround for server sending Content-Type: application/json with an empty body.
```
Comment (DXTimer) [inline dnsimple/dnsimple.go:269]: Please also create an internal issue so we can track this behavior in the API backend.

**Review by DXTimer — APPROVED** [review]
Left some comments, otherwise, this is good to go.

File: dnsimple/dnsimple.go (hunk 3 above)
Comment (wilson-dnsimple) [reply dnsimple/dnsimple.go:269]: Done: https://github.com/dnsimple/dnsimple-app/issues/23629

File: dnsimple/registrar_registrant.go
Comment (DXTimer) [inline dnsimple/registrar_registrant.go:1]: What do you think about renaming this to `registrar_registrant_changes.go`?

File: dnsimple/registrar_registrant.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
+package dnsimple
+
+import (
+	"context"
+	"fmt"
+)
+
+type CreateRegistrantChangeInput struct {
+	DomainId           int               `json:"domain_id"`
```
Comment (DXTimer) [inline dnsimple/registrar_registrant.go:9]: I realized that this should be a `string` type as our domain id could be also the actual domain name.

**Review by DXTimer — CHANGES_REQUESTED** [review]

**Review by DXTimer — APPROVED** [review]

File: dnsimple/registrar_registrant.go (hunk 4 above)
Comment (DXTimer) [reply dnsimple/registrar_registrant.go]: Fixed in [a0bab60](https://github.com/dnsimple/dnsimple-go/pull/146/commits/a0bab607f157c5a7a5600ef8036c3a8a7bbd4d4f)

**Pull request merged**